src/v2i-hub/PedestrianPlugin/src/FLIRWebSockAsyncClnSession.cpp

```diff
@@ -226,6 +261,10 @@ namespace PedestrianPlugin
                             int str_length_diff = 8 - idResult.length();
                             idResult.append(str_length_diff, '0');
 
+                            // Need to convert ID for SDSM
+                            // Map the value to be less than 65535
+                            int mappedID = static_cast<int>(id % 65535);
```

Comment:
What ID is this. I don't remember any none string or int IDs in the SDSM message.

Reply:
There are two IDs in the SDSM - sourceID(octet string) and objectID(long). I'm taking the ID provided from the FLIR for the objectID and since the value has to be less than 65535, I'm running a modulo on it.

While we're on it, I'm also using the same info from the FLIR as the sourceID. Should we let the user set the sourceID or can we randomize the ID so that the source and object aren't using the same data? The incoming pedestrian ID from FLIR will change for every new pedestrian, so the source would end up changing and we might not want that.

Reply:
I think the source ID is similar to the BSM ID and should be shared for all messages generated from V2X Hub. I think there may be other places were we write this ID. I would take a look at how we do it there. We probably just read it from a config parameter. Long term we want to pull this all from a single place like infrastructure ID

Reply:
Ok, for now I'm leaving it as being populated from the same FLIR pedestrian ID. We can look at using an Infrastructure ID or user input later.